src/platform/Darwin/BleConnectionDelegateImpl.mm

```diff
@@ -118,9 +123,13 @@ - (void)update;
 
         CHIP_ERROR BleConnectionDelegateImpl::CancelConnection()
         {
+            assertChipStackLockedByCurrentThread();
+
             ChipLogProgress(Ble, "%s", __FUNCTION__);
             if (ble) {
-                [ble stop];
+                dispatch_sync(ble.workQueue, ^{
```

Comment:
Does this have to be a sync dispatch?  It's probably OK because there are no other sync dispatches around here  but it's a little worrying because we do have sync dispatch to the Matter queue in other places, so if there is any way any of those APIs can get called while on the BLE work queue somehow we could get in trouble....

Reply:
I have updated the code such that the `BleConnection` instance is *only* accessed from the ble dedicated work queue.
As if, I have been able to rewrite those methods to be use an async dispatch. Let me know how it looks.